change-streams/main.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func ChangeStream(waitGroup sync.WaitGroup, stream *mongo.ChangeStream) {
```

Comment:
Right now, this function will execute indefinitely and there's no way to shut it down gracefully. One way to improve this would be to have it take a context and have the caller create a context using `context.WithCancel`. The returned `context.CancelFunc` can be used to signal the goroutine to exit.

Reply:
Are you suggesting something like this?:

```
func iterateChangeStream(routineCtx context.Context, waitGroup sync.WaitGroup, stream *mongo.ChangeStream) {
	defer stream.Close(routineCtx)
	defer waitGroup.Done()
	for stream.Next(routineCtx) {
		var data bson.M
		if err := stream.Decode(&data); err != nil {
			panic(err)
		}
		fmt.Printf("%v\n", data)
	}
	if stream.Err() != nil {
		panic(stream.Err())
	}
}

func main() {
        // ...
	routineCtx, cancelCtx := context.WithCancel(context.Background())
	go iterateChangeStream(routineCtx, waitGroup, episodesStream)

	// ...
}
```

Reply:
Yep. Only comment is that `cancelCtx` sounds like a context that can be cancelled. I'd call it something like `cancelFn` or `cancelCallback` to make it obvious that it's a function one can call to cancel the context.